src/components/Card.js

```diff
@@ -1,40 +1,71 @@
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
-    this._name = name;
-    this._link = link;
+  constructor(
+    data,
+    cardSelector,
+    handleImageClick,
+    handleDeleteCard,
+    _handleLikeButton,
+    handleCardLike,
+    handleCardDislike
+  ) {
+    this._name = data.name;
+    this._link = data.link;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteCard = handleDeleteCard;
+    this._id = data._id;
+    this._handleLikeButton = _handleLikeButton;
+    this.handleLike = handleCardLike;
+    this.handleDislike = handleCardDislike;
+    this._isLiked = data.isLiked;
+
+    this._cardElement = document
```

Comment:
Needs correction: you should create a card in the beginning of `getView` method, not here